staging/src/k8s.io/apiserver/pkg/endpoints/apiserver_test.go

```diff
@@ -3803,6 +3803,9 @@ func TestNamedCreaterWithGenerateName(t *testing.T) {
 	client := http.Client{}
 
 	simple := &genericapitesting.Simple{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "default",
+		},
```

Comment:
set this _after_ the `data, err := runtime.Encode` call that constructs the data we send to the server, to make sure we're not changing the test of what the server accepts. include a comment that we're expecting the namespace to be populated in the returned value

Reply:
same comment applies to the other places namespace was added to `simple` objects in this file

Reply:
@liggitt thanks for pointing that out.
I tried it out after your suggestion, the tests fail (wrt this PR, not https://github.com/kubernetes/kubernetes/pull/94637). I dug around a little and I think the reason for that is that https://github.com/kubernetes/kubernetes/pull/94637 makes changes to `createHandler` 
https://github.com/kubernetes/kubernetes/blob/2bca37340cb6d373b47d82db0b6c50b8ebc2234c/staging/src/k8s.io/apiserver/pkg/endpoints/handlers/create.go#L52
that sets the namespace (specifically: calling `EnsureObjectNamespaceMatchesRequestNamespace`)
In the current situation, `createHandler` has no such invocation. The place where the namespace would be set would be in `BeforeCreate` as part of the rest strategy 
https://github.com/kubernetes/kubernetes/blob/2bca37340cb6d373b47d82db0b6c50b8ebc2234c/staging/src/k8s.io/apiserver/pkg/registry/rest/create.go#L103
`ValidNamespace` then sets the ns
https://github.com/kubernetes/kubernetes/blob/2bca37340cb6d373b47d82db0b6c50b8ebc2234c/staging/src/k8s.io/apiserver/pkg/registry/rest/meta.go#L37

In the test, I don't _think_ strategy is involved because it is directly creating a handler and interacting with a server created using that handler. To verify, this is what I get when I add a breakpoint to `createHandler`
![Screenshot from 2022-02-16 21-06-01](https://user-images.githubusercontent.com/12381034/154299475-7587982c-bde8-4be7-bc1e-1c7d9bab595f.png)

On adding a breakpoint to `BeforeCreate` in strategy, it doesn't get reached.

Reply:
yeah, the change I described could only be made in the context of https://github.com/kubernetes/kubernetes/pull/94637, which updates the create and update handlers to do the namespace-setting as well

I pulled this commit into https://github.com/kubernetes/kubernetes/pull/94637 and addressed this comment in a second commit